src/BLSRegistryCoordinatorWithIndices.sol

```diff
@@ -502,6 +532,72 @@ contract BLSRegistryCoordinatorWithIndices is EIP712, Initializable, IBLSRegistr
         indexRegistry.deregisterOperator(operatorId, quorumNumbers);
     }
 
+    /**
+     * @dev perform the actual updates of the operators for each quorumNumber. This will revert if any of the operators
+     * are not registered for the quorum or if the number of operators does not match the total number of operators. By requiring
+     * ascending order of the operator addresses, there are no duplicate addresses and we can ensure that all operators are updated.
+     */
+    function _updateOperatorsForQuorum(
+        address[][] calldata operatorsPerQuorum,
+        bytes calldata quorumNumbers
+    ) internal {
+        for (uint256 i = 0; i < quorumNumbers.length; ++i) {
+            uint8 quorumNumber = uint8(quorumNumbers[i]);
+            address[] memory currQuorumOperators = operatorsPerQuorum[i];
+            require(
+                currQuorumOperators.length == indexRegistry.totalOperatorsForQuorum(quorumNumber),
+                "BLSRegistryCoordinatorWithIndices.updateOperatorsForQuorum: number of updated operators does not match quorum total"
+            );
+            address prevOperatorAddress = address(0);
+            // Update stakes for each operator in this quorum
+            for (uint256 j = 0; j < currQuorumOperators.length; ++j) {
+                address operator = currQuorumOperators[j];
+                OperatorInfo storage operatorInfo = _operatorInfo[operator];
+                bytes32 operatorId = operatorInfo.operatorId;
+                require(
+                    operatorInfo.status == OperatorStatus.REGISTERED,
+                    "BLSRegistryCoordinatorWithIndices.updateOperatorsForQuorum: Operator not registered"
+                );
+
+                {
+                    uint192 currentBitmap = _currentOperatorBitmap(operatorId);
```

Comment:
any way to reduce duplication between this and the other update stakes method?

Reply:
Let me try to create another PR for this

Reply:
https://github.com/Layr-Labs/eigenlayer-middleware/pull/68
Take a look here